src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-list/rbd-list.component.ts

```diff
@@ -195,14 +195,32 @@ export class RbdListComponent extends ListWithDetails implements OnInit {
         this.getInvalidNameDisable(selection) ||
         selection.first().schedule_info === undefined
     };
+    const promoteAction: CdTableAction = {
+      permission: 'update',
+      icon: Icons.edit,
+      click: () => this.actionPrimary(true),
+      name: this.actionLabels.PROMOTE,
+      visible: () =>
+        this.selection.first() != null && !this.selection.first().primary ? true : false
```

Comment:
The conditional clause already evaluates to a boolean:
```suggestion
        this.selection.first() != null && !this.selection.first().primary
```